From 0bc209ec46447b5e931275f1c5072b3c08ce256b Mon Sep 17 00:00:00 2001
From: Cory Fields <cory-nospam-@coryfields.com>
Date: Thu, 19 Jul 2018 14:21:05 -0400
Subject: [PATCH 5/6] build: avoid getifaddrs when unavailable

---
 configure.ac | 4 ++++
 src/net.cpp  | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ba13d1cd3..0123fdb11 100644
--- a/configure.ac
+++ b/configure.ac
@@ -608,6 +608,10 @@ fi
 
 AC_CHECK_HEADERS([endian.h sys/endian.h byteswap.h stdio.h stdlib.h unistd.h strings.h sys/types.h sys/stat.h sys/select.h sys/prctl.h])
 
+AC_CHECK_DECLS([getifaddrs, freeifaddrs],,,
+    [#include <sys/types.h>
+    #include <ifaddrs.h>]
+)
 AC_CHECK_DECLS([strnlen])
 
 # Check for daemon(3), unrelated to --with-daemon (although used by it)
diff --git a/src/net.cpp b/src/net.cpp
index 83bbaafb7..82ba53afb 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -2143,7 +2143,7 @@ void Discover(boost::thread_group& threadGroup)
             }
         }
     }
-#else
+#elif (HAVE_DECL_GETIFADDRS && HAVE_DECL_FREEIFADDRS)
     // Get local host ip
     struct ifaddrs* myaddrs;
     if (getifaddrs(&myaddrs) == 0)
-- 
2.11.0

